src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/resource/ReportDescriptionAddendumResource.kt

```diff
@@ -0,0 +1,265 @@
+      description = "The incident report id",
+      example = "11111111-2222-3333-4444-555555555555",
+      requiredMode = Schema.RequiredMode.REQUIRED,
+    )
+    @PathVariable
+    reportId: UUID,
+  ): List<DescriptionAddendum> {
+    return reportId.findReportOrThrowNotFound()
+      .descriptionAddendums.map { it.toDto() }
+  }
+
+  @PostMapping("/description-addendums")
+  @PreAuthorize("hasRole('ROLE_MAINTAIN_INCIDENT_REPORTS') and hasAuthority('SCOPE_write')")
+  @ResponseStatus(HttpStatus.CREATED)
+  @Operation(
+    summary = "Adds a description addendum to this incident report",
+    description = "Requires role MAINTAIN_INCIDENT_REPORTS and write scope. " +
+      "Authentication token must provide a username which is recorded as the report modifier and possibly as addendum author.",
+    responses = [
+      ApiResponse(
+        responseCode = "201",
+        description = "Returns description addendums",
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Invalid request",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Missing required role. Requires the MAINTAIN_INCIDENT_REPORTS role with write scope.",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Data not found",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  @Transactional
+  override fun addObject(
+    @Schema(
+      description = "The incident report id",
+      example = "11111111-2222-3333-4444-555555555555",
+      requiredMode = Schema.RequiredMode.REQUIRED,
+    )
+    @PathVariable
+    reportId: UUID,
+    @RequestBody
+    @Valid
+    request: AddDescriptionAddendum,
+  ): List<DescriptionAddendum> {
+    return reportId.updateReportOrThrowNotFound(
+      "Added description addendum to incident report",
+      WhatChanged.DESCRIPTION_ADDENDUMS,
+    ) { report ->
+      with(request) {
+        val sequence = if (report.descriptionAddendums.isEmpty()) 0 else report.descriptionAddendums.last().sequence + 1
+        report.addDescriptionAddendum(
+          sequence = sequence,
+          createdBy = createdBy ?: authenticationHolder.username ?: SYSTEM_USERNAME,
+          createdAt = createdAt ?: LocalDateTime.now(clock),
+          firstName = firstName,
+          lastName = lastName,
+          text = text,
+        )
+      }
+      report.descriptionAddendums.map { it.toDto() }
```

Comment:
you need to emit an event for any change

Reply:
the abstract super class does this. note the `updateReportOrThrowNotFound` on like 121 (and for all other object-modifying methods)

Ref: https://github.com/ministryofjustice/hmpps-incident-reporting-api/blob/70f5d3a230596448710482201788ef55080a5422/src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/resource/ReportRelatedObjectsResource.kt#L58-L64